### PR TITLE
esp32&esp32-c3/rt_timer: Re-word some comments and fix typos

### PR DESCRIPTION
**Pull request by Ouss4**
## Summary
 Re-word some comments and fix typos in `rt_timer` files in both ESP32 and ESP32-C3.
## Impact
No functional change.
## Testing
Build and test ESP32 and ESP32-C3 defconfigs.

### CODE REVIEW

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c [hunk 1]
```diff
@@ -81,13 +81,13 @@ static struct esp32c3_tim_dev_s *s_esp32c3_tim_dev;
  * Name: start_rt_timer
  *
  * Description:
- *   Start timer by inserting it into running list and reset hardware timer
- *   alarm value if this timer in head of list.
+ *   Start the timer by inserting it into the running list and reset the
+ *   hardware timer alarm value if this timer in at the head of the list.
```
Comment (davids5) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:85]:
```suggestion
 *   hardware timer alarm value if this timer is at the head of the list.
```

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c [hunk 2]
```diff
@@ -212,15 +213,13 @@ static void stop_rt_timer(FAR struct rt_timer_s *timer)
...
+      /* If the timer is at the head of the list */
 
       if (ishead)
         {
-          /* If list is not empty */
-
           if (!list_is_empty(&s_runlist))
             {
-              /* Reset hardware timer alarm value to be next timer's */
+              /* Reset the hardware timer alarm value to be next timer's */
```
Comment (davids5) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:222]:
```suggestion
              /* Reset the hardware timer alarm value to be next timer's timeout or alarm ??? */
```

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c [hunk 3]
```diff
@@ -240,9 +239,9 @@ static void stop_rt_timer(FAR struct rt_timer_s *timer)
  * Name: delete_rt_timer
  *
  * Description:
- *   Delete timer by removing it from list, then set the timer's state
- *   to be "RT_TIMER_DELETE", inserting into work list to let rt-timer
- *   thread to delete it and free resource.
+ *   Delete the timer by removing it from the list, then set the timer's
+ *   state to be "RT_TIMER_DELETE", finally insert it into the work list
```
Comment (davids5) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:243]:
```suggestion
 *   state to "RT_TIMER_DELETE", and finally insert it into the work list
```

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c [hunk 4]
```diff
@@ -291,8 +290,8 @@ static void delete_rt_timer(FAR struct rt_timer_s *timer)
  * Name: rt_timer_thread
  *
  * Description:
- *   RT timer working thread, it wait for a timeout semaphore, scan
- *   the timeout list and process all timers in this list.
+ *   RT timer working thread, it waits for a timeout semaphore, scans
```
Comment (davids5) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:293]:
```suggestion
 *   RT timer working thread: Waits for a timeout semaphore, scans
```

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c [hunk 5]
```diff
@@ -425,20 +420,19 @@ static int rt_timer_isr(int irq, void *context, void *arg)
       ESP32C3_TIM_GETCTR(tim, &counter);
       if (timer->alarm <= counter)
         {
-          /**
-           * Remove first timer in running list and add it into
-           * timeout list.
+          /* Remove the first timer in the running list and add it to
```
Comment (davids5) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:423]:
```suggestion
          /* Remove the first timer from the running list and add it to
```

File: arch/xtensa/src/esp32/esp32_rt_timer.c [hunk 6]
```diff
@@ -80,13 +80,13 @@ static struct esp32_tim_dev_s *s_esp32_tim_dev;
  * Name: start_rt_timer
  *
  * Description:
- *   Start timer by inserting it into running list and reset hardware timer
- *   alarm value if this timer in head of list.
+ *   Start the timer by inserting it into the running list and reset the
+ *   hardware timer alarm value if this timer in at the head of the list.
```
Comment (davids5) [inline arch/xtensa/src/esp32/esp32_rt_timer.c:84]:
```suggestion
 *   hardware timer alarm value if this timer is at the head of the list.
```

File: arch/xtensa/src/esp32/esp32_rt_timer.c [hunk 7]
```diff
@@ -239,9 +238,9 @@ static void stop_rt_timer(FAR struct rt_timer_s *timer)
  * Name: delete_rt_timer
  *
  * Description:
- *   Delete timer by removing it from list, then set the timer's state
- *   to be "RT_TIMER_DELETE", inserting into work list to let rt-timer
- *   thread to delete it and free resource.
+ *   Delete the timer by removing it from the list, then set the timer's
+ *   state to be "RT_TIMER_DELETE", finally insert it into the work list
```
Comment (davids5) [inline arch/xtensa/src/esp32/esp32_rt_timer.c:242]:
```suggestion
 *   state to "RT_TIMER_DELETE", and finally insert it into the work list
```

**Review by davids5 — CHANGES_REQUESTED** [review]
Some minor edits

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c [hunk 8]
```diff
@@ -187,18 +189,17 @@ static void stop_rt_timer(FAR struct rt_timer_s *timer)
 
   flags = enter_critical_section();
 
-  /**
-   * Function "start" can set timer to be repeat, and function "stop"
-   * should remove this feature although it is not in ready state.
+  /* "start" function can set the timer's repeat flag, and function "stop"
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:192]:
```suggestion
  /* "start" function can set the timer's repeat flag, and "stop" function
```

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c (hunk 2 above)
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:222]: I believe the current writing is correct, but here is a suggestion for a (maybe) more direct explanation:
```suggestion
              /* Set the value from next timer as the new hardware timer alarm value */
```

File: arch/xtensa/src/esp32/esp32_rt_timer.c [hunk 9]
```diff
@@ -397,13 +392,13 @@ static int rt_timer_isr(int irq, void *context, void *arg)
 
   ESP32_TIM_ACKINT(tim);
 
-  /* Wake up thread to process timeout timers */
+  /* Wake up the thread to process timeout timers */
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32/esp32_rt_timer.c:395]:
```suggestion
  /* Wake up the thread to process timed-out timers */
```

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c [hunk 10]
```diff
@@ -451,14 +445,14 @@ static int rt_timer_isr(int irq, void *context, void *arg)
...
-      /* If there is timer in list, alarm should be enable */
+      /* If there is a timer in the list, the alarm should be enabled */
 
       ESP32C3_TIM_SETALRM(tim, true);
     }
 
   if (wake)
     {
-      /* Wake up thread to process timeout timers */
+      /* Wake up the thread to process timeout timers */
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:455]:
```suggestion
      /* Wake up the thread to process timed-out timers */
```

File: arch/risc-v/src/esp32c3/esp32c3_rt_timer.c [hunk 11]
```diff
@@ -312,7 +311,7 @@ static int rt_timer_thread(int argc, FAR char *argv[])
 
   while (1)
     {
-      /* Waiting for timers timeout */
+      /* Waiting for the timers timeout */
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_rt_timer.c:314]:
```suggestion
      /* Waiting for all timers to time out */
```

**Comment by Ouss4** [comment]
@gustavonihei @davids5 thanks for your review, I applied all the suggestions.

File: arch/xtensa/src/esp32/esp32_rt_timer.c [hunk 12]
```diff
@@ -302,7 +303,7 @@ static int rt_timer_thread(int argc, FAR char *argv[])
 
   while (1)
     {
-      /* Waiting for timers timeout */
+      /* Waiting for the timers timeouts */
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32/esp32_rt_timer.c:306]:
```suggestion
      /* Waiting for all timers to time out */
```
Suggestion to apply the same modification from other file.

File: arch/xtensa/src/esp32/esp32_rt_timer.c [hunk 13]
```diff
@@ -281,8 +282,8 @@ static void delete_rt_timer(FAR struct rt_timer_s *timer)
  * Name: rt_timer_thread
  *
  * Description:
- *   RT timer working thread, it wait for a timeout semaphore, scan
- *   the timeout list and process all timers in this list.
+ *   RT timer working thread, it waits for a timeout semaphore, scans
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32/esp32_rt_timer.c:285]:
```suggestion
 *   RT timer working thread: Waits for a timeout semaphore, scans
```
Copying the modification from the other file.

**Review by gustavonihei — APPROVED** [review]

**Pull request merged**